packages/dart/noports_core/lib/src/srv/srv_impl.dart

```diff
@@ -880,18 +1049,56 @@ class SrvImplDart implements Srv<SocketConnector> {
     }
     InternetAddress localAddress = await resolveRequestedLocalHost();
 
-    SocketConnector socketConnector = await SocketConnector.socketToSocket(
-        addressA: localAddress,
-        portA: localPort,
-        addressB: relayAddress,
-        portB: streamingPort,
-        verbose: Platform.environment['SRV_TRACE'] == 'true',
-        logger: ioSinkForLogger(logger),
-        transformAtoB: encrypter,
-        transformBtoA: decrypter);
-    if (rvdAuthString != null) {
+    bool verbose = Platform.environment['SRV_TRACE'] == 'true';
+    IOSink logSink = ioSinkForLogger(logger);
+    SocketConnector socketConnector = SocketConnector(
+      verbose: verbose,
+      logger: ioSinkForLogger(logger),
+    );
+    if (verbose) {
+      logSink.writeln('socket_connector:'
+          ' Connecting to $localAddress:$localPort');
+    }
+    Socket sideASocket = await Socket.connect(localAddress, localPort);
```

Comment:
For single sessions (i.e. sshnp) we try to connect locally first and give up if that doesn't work

Reply:
For "multi" sessions (i.e. npt) (see earlier in _daemonSideMulti stuff) we try to connect to the relay first, and then to local. We do this because of potential timing issues where a local socket might just time out after we connect, while we're still authenticating to the relay. (I saw this during testing, hence I switched the order for 'multi' to relay-first, local-second